src/_pytest/runner.py

```diff
@@ -200,9 +203,15 @@ def check_interactive_exception(call, report):
     )
 
 
-def call_runtest_hook(item, when, **kwds):
-    hookname = "pytest_runtest_" + when
-    ihook = getattr(item.ihook, hookname)
+def call_runtest_hook(item, when: "Literal['setup', 'call', 'teardown']", **kwds):
+    if when == "setup":
+        ihook = item.ihook.pytest_runtest_setup
+    elif when == "call":
+        ihook = item.ihook.pytest_runtest_call
+    elif when == "teardown":
+        ihook = item.ihook.pytest_runtest_teardown
+    else:
+        assert False, "Unhandled runtest hook case: {}".format(when)
```

Comment:
Do you like to go with `assert False` in .coveragerc then?
Please add it here then maybe already.

Reply:
Sure, added a commit for this.